File: baekjoon/11724.java
import java.nio.charset.StandardCharsets;
import java.util.*;

public class Main
{
    static int N, M;
    static ArrayList<Integer>[] edge = new ArrayList[1010];
    static boolean[] mark = new boolean[1010];

    static void dfs(int v)
    {
        if(mark[v]) return;
        mark[v] = true;
        for(int x : edge[v]) dfs(x);
    }

    public static void main(String[] args)
    {
        int cnt = 0;
        Scanner sc = new Scanner(System.in);
        for(int i = 0; i < 1010; i++) edge[i] = new ArrayList<Integer>();
        N = sc.nextInt(); M = sc.nextInt();

        for(int i = 0; i < M; i++)
        {
            int a = sc.nextInt();
            int b = sc.nextInt();
            edge[a].add(b);
            edge[b].add(a);
        }

        for(int i = 1; i <= N; i++)
        {
            if(mark[i]) continue;
            dfs(i);
            cnt++;
        }
        System.out.print(cnt);
    }
}

File: baekjoon/11724.kt
import java.util.*

var edge = Array<ArrayList<Int>>(1010) {ArrayList<Int>(0)}
var mark = Array<Boolean>(1010) { false }
var N = 0
var M = 0

fun dfs(v: Int)
{
    if(mark[v]) return
    mark[v] = true
    for(x in edge[v]) dfs(x)
}

fun main()
{
    val sc = Scanner(System.`in`)
    N = sc.nextInt(); M = sc.nextInt()
    var cnt = 0

    for(i in 0 until M)
    {
        var a = sc.nextInt()
        var b = sc.nextInt()
        edge[a].add(b)
        edge[b].add(a)
    }

    for(i in 1..N)
    {
        if(mark[i]) continue
        dfs(i)
        cnt++
    }
    print(cnt)
}

